includes/admin/tools/data/class-give-tools-delete-donations-only.php

```diff
@@ -0,0 +1,323 @@
+<?php
+/**
+ * Delete Transactions between a date range.
+ *
+ * This class handles batch processing of deleting transactions between a given range.
+ *
+ * @package     Admin/Tools
+ * @subpackage  Admin/Tools/Give_Tools_Delete_Donations_Only
+ * @copyright   Copyright (c) 2016, WordImpress
+ * @license     https://opensource.org/licenses/gpl-license GNU Public License
+ * @since       2.3.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Give_Tools_Delete_Donations_Only Class
+ *
+ * @since 2.3.0
+ */
+class Give_Tools_Delete_Donations_Only extends Give_Batch_Export {
+
+	/**
+	 * Our export type. Used for export-type specific filters/actions.
+	 *
+	 * @var string
+	 * @since 2.3.0
+	 */
+	public $export_type = '';
+
+	/**
+	 * Allows for a non-form batch processing to be run.
+	 *
+	 * @since 2.3.0
+	 * @var boolean
+	 */
+	public $is_void = true;
+
+	/**
+	 * Sets the number of items to pull on each step.
+	 *
+	 * @since 2.3.0
+	 * @var integer
+	 */
+	public $per_step = 30;
+
+	/**
+	 * Set the start date of donation.
+	 *
+	 * @since 2.3.0
+	 * @var string
+	 */
+	public $start_date = '';
+
+	/**
+	 * Set the end date of donation.
+	 *
+	 * @since 2.3.0
+	 * @var string
+	 */
+	public $end_date = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param number $_step Step ID of the currenct batch.
+	 */
+	public function __construct( $_step = 1 ) {
+		parent::__construct( $_step );
+
+		$this->is_writable = true;
+	}
+
+	/**
+	 * Get the Export Data
+	 *
+	 * @access public
+	 * @since 2.3.0
+	 * @global object $wpdb Used to query the database using the WordPress Database API
+	 *
+	 * @return array|bool $data The data for the CSV file
+	 */
+	public function get_data() {
+		$items = $this->get_stored_data( 'give_temp_delete_donation_ids' );
+
+		if ( ! is_array( $items ) ) {
+			return false;
+		}
+
+		$offset     = ( $this->step - 1 ) * $this->per_step;
+		$step_items = array_slice( $items, $offset, $this->per_step );
+		$meta_table = __give_v20_bc_table_details( 'payment' );
+
+		if ( $step_items ) {
+			foreach ( $step_items as $item ) {
+				// Delete the main payment.
+				give_delete_donation( absint( $item['id'] ) );
+			}
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Return the calculated completion percentage
+	 *
+	 * @since 2.3.0
+	 * @return int
+	 */
+	public function get_percentage_complete() {
+
+		$items = $this->get_stored_data( 'give_temp_delete_donation_ids', false );
+		$total = count( $items );
```

Comment:
@Sidsector9 when donations is empty then I am getting this error:

`PHP Warning:  count(): Parameter must be an array or an object that implements Countable in /app/public/wp-content/plugins/Give/includes/admin/tools/data/class-give-tools-delete-donations-only.php on line 117`